From 517cd3032fa424c6b79e26858b0457cb5bfae7dd Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Wed, 3 Apr 2024 16:29:12 +0200
Subject: Revert "workqueue: Don't call cpumask_test_cpu() with -1 CPU in wq_update_node_max_active()"

From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

This reverts commit 9fc557d489f8163c1aabcb89114b8eba960f4097 which is commit
15930da42f8981dc42c19038042947b475b19f47 upstream.

Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 kernel/workqueue.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/kernel/workqueue.c
+++ b/kernel/workqueue.c
@@ -1506,7 +1506,7 @@ static void wq_update_node_max_active(st
 
 	lockdep_assert_held(&wq->mutex);
 
-	if (off_cpu >= 0 && !cpumask_test_cpu(off_cpu, effective))
+	if (!cpumask_test_cpu(off_cpu, effective))
 		off_cpu = -1;
 
 	total_cpus = cpumask_weight_and(effective, cpu_online_mask);
